From b4695b946739e10d44fb5f5e6df1a7858dbde987 Mon Sep 17 00:00:00 2001
From: Frederik Schwan <frederik@tty42.de>
Date: Wed, 2 Apr 2025 22:27:15 +0200
Subject: [PATCH] add compatibility changes to allow both 64 bit and 32 bit
 versions of the package to be installed at the same time

---
 etc/pcscd.service.in | 10 +++++-----
 meson.build          |  7 ++++---
 meson.options        |  5 +++++
 3 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/etc/pcscd.service.in b/etc/pcscd.service.in
index 3c5fc7f7..bafe9356 100644
--- a/etc/pcscd.service.in
+++ b/etc/pcscd.service.in
@@ -1,13 +1,13 @@
 [Unit]
 Description=PC/SC Smart Card Daemon
-Requires=pcscd.socket
+Requires=pcscd@program_suffix@.socket
 @requires_polkit@
 Documentation=man:pcscd(8)
 
 [Service]
-ExecStart=@sbindir_exp@/pcscd --foreground --auto-exit $PCSCD_ARGS
-ExecReload=@sbindir_exp@/pcscd --hotplug
-EnvironmentFile=-@sysconfdir@/default/pcscd
+ExecStart=@sbindir_exp@/pcscd@program_suffix@ --foreground --auto-exit $PCSCD_ARGS
+ExecReload=@sbindir_exp@/pcscd@program_suffix@ --hotplug
+EnvironmentFile=-@sysconfdir@/default/pcscd@program_suffix@
 
 # Paths
 ProtectProc=invisible
@@ -44,4 +44,4 @@ SystemCallFilter=~@resources @privileged
 SystemCallArchitectures=native
 
 [Install]
-Also=pcscd.socket
+Also=pcscd@program_suffix@.socket
diff --git a/meson.build b/meson.build
index f07bc1c9..413334f2 100644
--- a/meson.build
+++ b/meson.build
@@ -29,6 +29,7 @@ confgen_data = configuration_data({
   'sbindir_exp' : sbindir,
   'PCSCLITE_CONFIG_DIR' : get_option('serialconfdir'),
   'usbdropdir' : get_option('usbdropdir'),
+  'program_suffix' : get_option('program_suffix'),
   })
 
 # tests for functions
@@ -153,7 +154,7 @@ features += 'serialconfdir=' + get_option('serialconfdir')
 # generate PCSCLITE_FEATURES
 conf_data.set('PCSCLITE_FEATURES', '"' + ' '.join(features) + '"')
 
-executable('pcscd',
+executable('pcscd@0@'.format(get_option('program_suffix')),
   sources : pcscd_src,
   include_directories : incdir,
   dependencies : pcscd_dep,
@@ -285,11 +286,11 @@ if get_option('libsystemd')
   systemd = dependency('systemd')
   unit = get_option('systemdunit')
   systemdsystemunitdir = systemd.get_variable(pkgconfig : 'systemd' + unit + 'unitdir')
-configure_file(output : 'pcscd.socket',
+configure_file(output : 'pcscd@0@.socket'.format(get_option('program_suffix')),
   input : 'etc/pcscd.socket.in',
   install_dir : systemdsystemunitdir,
   configuration : confgen_data)
-configure_file(output : 'pcscd.service',
+configure_file(output : 'pcscd@0@.service'.format(get_option('program_suffix')),
   input : 'etc/pcscd.service.in',
   install_dir : systemdsystemunitdir,
   configuration : confgen_data)
diff --git a/meson.options b/meson.options
index f0b58cac..9e62bcc1 100644
--- a/meson.options
+++ b/meson.options
@@ -53,3 +53,8 @@ option('filter_names',
   type : 'boolean',
   value : true,
   description : 'reader filtering using PCSCLITE_FILTER_IGNORE_READER_NAMES and PCSCLITE_FILTER_EXTEND_READER_NAMES')
+
+option('program_suffix',
+  type : 'string',
+  value : '',
+  description : 'suffix for programs')
-- 
2.49.0

